core/src/transport/mod.rs

```diff
@@ -54,6 +54,36 @@ pub use self::memory::connector;
 pub use self::muxed::MuxedTransport;
 pub use self::upgrade::UpgradedNode;
 
+
+// This is a new error type that you've created. It represents the ways a
+// toolchain could be invalid.
+//
+// The custom derive for Fail derives an impl of both Fail and Display.
+// We don't do any other magic like creating new types.
+#[derive(Debug, Fail)]
+pub enum TransportError {
+    #[fail(display= "Error parsing Address {}: {}", addr, err)]
+    MultiAddrParseError {
```

Comment:
That shouldn't be a variant, as we directly pass a `Multiaddr`.

Reply:
Though I intended to have this, I ended up not using this variant. I'm removing it.